central/cluster/datastore/datastore_impl.go

```diff
@@ -353,6 +354,10 @@ func (ds *datastoreImpl) addClusterNoLock(ctx context.Context, cluster *storage.
 	if cluster.GetName() == "" {
 		return "", errors.New("cannot add a cluster without name")
 	}
+	if valid, err := pkgCluster.IsNameValid(cluster.GetName()); !valid {
+		return "", errors.Wrap(err, "invalid cluster name")
```

Comment:
```suggestion
		return "", errox.InvalidArgs.CausedBy(err)
```
to trigger `404` instead of 500.